test/parallel/test-tls-keylog-tlsv13.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const events = require('events');
+const tls = require('tls');
+const fixtures = require('../common/fixtures');
+
+async function main() {
+  // Create server / client pair
+  const server = tls.createServer({
+    key: fixtures.readSync('/keys/agent2-key.pem'),
+    cert: fixtures.readSync('/keys/agent2-cert.pem'),
+    minVersion: 'TLSv1.3',
+    maxVersion: 'TLSv1.3',
+  });
+  server.listen();
+  await events.once(server, 'listening');
+  server.once('connection', () => server.close());
+
+  const client = tls.connect({
+    port: server.address().port,
+    rejectUnauthorized: false,
+  });
+
+  // Record keylog lines
+  const lines = { client: new Set(), server: new Set() };
+  client.on('keylog', (line) => lines.client.add(line));
+  server.on('keylog', (line) => lines.server.add(line));
+
+  // Wait for handshake, test that same lines were produced
+  await Promise.all([ events.once(server, 'secureConnection'),
+                      events.once(client, 'secureConnect') ]);
+  assert.deepStrictEqual(lines.client, lines.server);
+  assert.strictEqual(lines.client.size, 5);
+
+  client.destroy();
```

Comment:
It looks like this test is flaky or has platform-dependent behaviour, because it fails in CI with `ECONNRESET` on in some runs, e.g.: https://ci.nodejs.org/job/node-test-commit-freebsd/26170/nodes=freebsd11-x64/testReport/junit/(root)/test/parallel_test_tls_keylog_tlsv13/

Without having tried it myself, I think using `.end()` would also work and might solve that issue, or, if it doesn’t, installing a dummy `.on('error')` handler?

Reply:
these tests tend to be more reliable when they use .end(), as suggested, particularly if they pump a single byte of data from client to server, and back, before server.close.

Reply:
The test now calls `server.close()` and `client.end()`, which is the behaviour of the fixture pointed out by @sam-github. Let's see now.

Reply:
The test seems to be reliable now, so I'm marking this as resolved :)